docs/data/charts/legend/LegendClickNoSnap.js

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import Stack from '@mui/material/Stack';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import IconButton from '@mui/material/IconButton';
+import UndoOutlinedIcon from '@mui/icons-material/UndoOutlined';
+
+import { ChartsLegend, PiecewiseColorLegend } from '@mui/x-charts/ChartsLegend';
+
+import { HighlightedCode } from '@mui/docs/HighlightedCode';
+import { ResponsiveChartContainer } from '@mui/x-charts/ResponsiveChartContainer';
+
+const pieSeries = [
+  {
+    type: 'pie',
+    id: 'series-1',
+    label: 'Series 1',
+    data: [
+      { label: 'Pie A', id: 'P1-A', value: 400 },
+      { label: 'Pie B', id: 'P2-B', value: 300 },
+    ],
+  },
+];
+
+const barSeries = [
+  {
+    type: 'bar',
+    id: 'series-1',
+    label: 'Series 1',
+    data: [0, 1, 2],
+  },
+  {
+    type: 'bar',
+    id: 'series-2',
+    label: 'Series 2',
+    data: [0, 1, 2],
+  },
+];
+
+const lineSeries = [
+  {
+    type: 'line',
+    id: 'series-1',
+    label: 'Series 1',
+    data: [0, 1, 2],
+  },
+  {
+    type: 'line',
+    id: 'series-2',
+    label: 'Series 2',
+    data: [0, 1, 2],
+  },
+];
+
+export default function LegendClickNoSnap() {
+  const [itemData, setItemData] = React.useState();
+
+  return (
+    <Stack
+      direction={{ xs: 'column', md: 'row' }}
+      spacing={{ xs: 0, md: 4 }}
+      sx={{ width: '100%' }}
+    >
+      <Box sx={{ flexGrow: 1 }}>
+        <Typography>Chart Legend</Typography>
+        <ResponsiveChartContainer series={barSeries} width={400} height={60}>
+          <ChartsLegend
+            direction="row"
+            position={{
+              horizontal: 'left',
+              vertical: 'top',
+            }}
+            slotProps={{
+              legend: {
+                onItemClick: (event, context, index) =>
+                  setItemData([context, index]),
+              },
+            }}
+          />
+        </ResponsiveChartContainer>
+        <Typography>Pie Chart Legend</Typography>
+        <ResponsiveChartContainer series={pieSeries} width={400} height={60}>
+          <ChartsLegend
+            direction="row"
+            position={{
+              horizontal: 'left',
+              vertical: 'top',
+            }}
+            slotProps={{
+              legend: {
+                onItemClick: (event, context, index) =>
+                  setItemData([context, index]),
+              },
+            }}
+          />
```

Comment:
I believe this to be a mistake on the `ChartsLegend` component. 

The color legend can accept the `onClick` event as a prop, while the `ChartsLegend` need to be passed as a prop

```tsx
<PiecewiseColorLegend
  onItemClick={(event, context, index) => setItemData([context, index])}
/>

// vs

<ChartsLegend
  slotProps={{
    legend: {
      onItemClick: (event, context, index) => setItemData([context, index]),
    },
  }}
/>
```

Reply:
Effectively this makes no sens

```
<ChartsLegend
  slotProps={{
    legend: {
      onItemClick: (event, context, index) => setItemData([context, index]),
    },
  }}
/>
```

Reply:
Yep, that is what I meant 😅

Reply:
The issue is that `<ChartsLegend />` is doing some preprocessing before calling the `slot.legend` with those data. But nothing prevent us from adding the `onItemClick` to the `ChartsLegend` and propagate it accordingly